pkg/cmd/pr/merge/merge.go

```diff
@@ -253,37 +269,50 @@ func (m *mergeContext) merge() error {
 		setCommitBody: m.opts.BodySet,
 	}
 
+	if m.addToMergeQueue() && !m.opts.InteractiveMode {
+		_ = m.warnf("%s A merge strategy cannot be set when adding a pull request to the merge queue\n", m.cs.FailureIconWithColor(m.cs.Red))
```

Comment:
What does this error condition help with? Should this always error for merge queue case when running in non-interactive mode? How would a person non-interactively add a PR to merge queue then?

Reply:
the merge strategy is configured on the branch protection for the queue. if the user provides a merge strategy it will be ignored by the queue.

it's more accurate to say that at this point the user is not merging, they are only adding to the queue, and the queue will merge later on behalf of the user.

Reply:
I'm not sure I understand, since this condition doesn't seem to test for a specific merge strategy. It seems that this error will always get thrown in non-interactive merge queue mode, regardless of strategy. Or am I missing some other condition before this?